### PR TITLE
trivial: Fix Python3 collections

### PR DESCRIPTION
**Pull request by homalozoa**
1. collections.Mapping, collections.Hashable, collections.Iterable, etc are deprecated in Python3.3, removed in Python3.10. Ref to: [collections.abc](https://docs.python.org/3/library/collections.abc.html)
2. Fix style for Python3 style check

Signed-off-by: Homalozoa <xuhaiwang@xiaomi.com>

### CODE REVIEW

**Comment by axel-h** [comment]
The commit message mentions a style change also, but I can't see anything related to this in the diff?

**Comment by homalozoa** [comment]
It seems failed to sync my commit. I'll try later.

**Comment by axel-h** [comment]
Please make the style changes a separate commit.

**Comment by homalozoa** [comment]
Okay. No problem.

File: camkes/ast/liftedast.py [hunk 1]
```diff
@@ -57,14 +59,15 @@ def assembly(self):
 
     def filter(self, function=None):
         if function is None:
-            function = lambda x: x is not None
+            def function(x): return x is not None
```
Comment (axel-h) [inline camkes/ast/liftedast.py:62]: This is more then just a style change, please make this also a separate commit.

Comment (homalozoa) [reply camkes/ast/liftedast.py:62]: Done.

File: camkes/ast/liftedast.py [hunk 2]
```diff
@@ -59,7 +59,7 @@ def assembly(self):
...
```
Comment (axel-h) [inline camkes/ast/liftedast.py:62]: In the old code `function` was a variable holding a function or `None`. In your change a local function named `function` is introduced that shadows the actual variable. I'm not sure how much difference this makes in Python in the end, but the cleaner way would be be:
```
def my_function(x): return x is not None
function = my_function
```
The original code was basically a shorter version of this, using `lambda` to create an anonymous function. Not sure what "ref E731" in the commit message is supposed to mean, I could not relate this to anything. Also, while some time back there have been attempts to deprecate `lambda`, it well accepted now, see https://mail.python.org/pipermail/python-dev/2006-February/060415.html

Comment (axel-h) [reply camkes/ast/liftedast.py:62]: ah ok, found it at https://www.flake8rules.com/rules/E731.html
But still, we should not shadow the variable here.

Comment (homalozoa) [reply camkes/ast/liftedast.py:62]: Yes. It's a flake 8 rule.

> In the old code `function` was a variable holding a function or `None`. In your change a local function named `function` is introduced that shadows the actual variable. I'm not sure how much difference this makes in Python in the end, but the cleaner way would be be:
> 
> ```
> def my_function(x): return x is not None
> function = my_function
> ```
> 
> The original code was basically a shorter version of this, using `lambda` to create an anonymous function. Not sure what "ref E731" in the commit message is supposed to mean, I could not relate this to anything. Also, while some time back there have been attempts to deprecate `lambda`, it well accepted now, see https://mail.python.org/pipermail/python-dev/2006-February/060415.html

Let me change it.

**Review by axel-h — APPROVED** [review]

**Review by kent-mcleod — APPROVED** [review]

**Comment by kent-mcleod** [comment]
Thanks!
